Exception with Postgres BYTEA type

local-data-api image: koxudaxi/local-data-api:0.5.4
PostgreSQL image: postgres:10.12-alpine

Given table such as
```sql
			CREATE TABLE IF NOT EXISTS events (
				id SERIAL PRIMARY KEY,
				rep VARCHAR(64),
				data BYTEA
			)
```
and a query such as
```sql
SELECT id, data FROM events
```
local-data-api throws this exception:
```
local-data-api_1  | 192.168.144.1:45132 - "POST /Execute HTTP/1.1" 500
local-data-api_1  | [2021-03-04 16:48:35 +0000] [7] [ERROR] Exception in ASGI application
local-data-api_1  | Traceback (most recent call last):
local-data-api_1  |   File "/usr/local/lib/python3.8/site-packages/uvicorn/protocols/http/httptools_impl.py", line 396, in run_asgi
local-data-api_1  |     result = await app(self.scope, self.receive, self.send)
local-data-api_1  |   File "/usr/local/lib/python3.8/site-packages/uvicorn/middleware/proxy_headers.py", line 45, in __call__
local-data-api_1  |     return await self.app(scope, receive, send)
local-data-api_1  |   File "/usr/local/lib/python3.8/site-packages/fastapi/applications.py", line 140, in __call__
local-data-api_1  |     await super().__call__(scope, receive, send)
local-data-api_1  |   File "/usr/local/lib/python3.8/site-packages/starlette/applications.py", line 134, in __call__
local-data-api_1  |     await self.error_middleware(scope, receive, send)
local-data-api_1  |   File "/usr/local/lib/python3.8/site-packages/starlette/middleware/errors.py", line 178, in __call__
local-data-api_1  |     raise exc from None
local-data-api_1  |   File "/usr/local/lib/python3.8/site-packages/starlette/middleware/errors.py", line 156, in __call__
local-data-api_1  |     await self.app(scope, receive, _send)
local-data-api_1  |   File "/usr/local/lib/python3.8/site-packages/starlette/exceptions.py", line 73, in __call__
local-data-api_1  |     raise exc from None
local-data-api_1  |   File "/usr/local/lib/python3.8/site-packages/starlette/exceptions.py", line 62, in __call__
local-data-api_1  |     await self.app(scope, receive, sender)
local-data-api_1  |   File "/usr/local/lib/python3.8/site-packages/starlette/routing.py", line 590, in __call__
local-data-api_1  |     await route(scope, receive, send)
local-data-api_1  |   File "/usr/local/lib/python3.8/site-packages/starlette/routing.py", line 208, in __call__
local-data-api_1  |     await self.app(scope, receive, send)
local-data-api_1  |   File "/usr/local/lib/python3.8/site-packages/starlette/routing.py", line 41, in app
local-data-api_1  |     response = await func(request)
local-data-api_1  |   File "/usr/local/lib/python3.8/site-packages/fastapi/routing.py", line 129, in app
local-data-api_1  |     raw_response = await run_in_threadpool(dependant.call, **values)
local-data-api_1  |   File "/usr/local/lib/python3.8/site-packages/starlette/concurrency.py", line 25, in run_in_threadpool
local-data-api_1  |     return await loop.run_in_executor(None, func, *args)
local-data-api_1  |   File "/usr/local/lib/python3.8/concurrent/futures/thread.py", line 57, in run
local-data-api_1  |     result = self.fn(*self.args, **self.kwargs)
local-data-api_1  |   File "/app/local_data_api/main.py", line 99, in execute_statement
local-data-api_1  |     response: ExecuteStatementResponse = resource.execute(
local-data-api_1  |   File "/app/local_data_api/resources/jdbc/__init__.py", line 158, in execute
local-data-api_1  |     records=[
local-data-api_1  |   File "/app/local_data_api/resources/jdbc/__init__.py", line 159, in <listcomp>
local-data-api_1  |     [
local-data-api_1  |   File "/app/local_data_api/resources/jdbc/__init__.py", line 160, in <listcomp>
local-data-api_1  |     self.get_filed_from_jdbc_type(
local-data-api_1  |   File "/app/local_data_api/resources/jdbc/postgres.py", line 45, in get_filed_from_jdbc_type
local-data-api_1  |     return super().get_filed_from_jdbc_type(value, jdbc_type)
local-data-api_1  |   File "/app/local_data_api/resources/jdbc/__init__.py", line 97, in get_filed_from_jdbc_type
local-data-api_1  |     return Field(blobValue=b64encode(value))
local-data-api_1  |   File "/usr/local/lib/python3.8/base64.py", line 58, in b64encode
local-data-api_1  |     encoded = binascii.b2a_base64(s, newline=False)
local-data-api_1  | TypeError: a bytes-like object is required, not 'str'
```

--- a/local_data_api/resources/jdbc/__init__.py
+++ b/local_data_api/resources/jdbc/__init__.py
@@ -94,6 +94,8 @@ class JDBC(Resource, ABC):
             elif type_ in BOOLEAN:
                 return Field(booleanValue=value)
             elif type_ in BLOB:
+                if isinstance(value, str):
+                    value = value.encode()
                 return Field(blobValue=b64encode(value))
 
         return self.get_field_from_value(value)

--- a/local_data_api/resources/jdbc/__init__.py
+++ b/local_data_api/resources/jdbc/__init__.py
@@ -94,7 +94,7 @@ class JDBC(Resource, ABC):
             elif type_ in BOOLEAN:
                 return Field(booleanValue=value)
             elif type_ in BLOB:
-                if isinstance(value, str):
+                if isinstance(value, str):  # pragma: no cover
                     value = value.encode()
                 return Field(blobValue=b64encode(value))
 

--- a/scripts/integration-test.sh
+++ b/scripts/integration-test.sh
@@ -58,6 +58,15 @@ function test {
         --include-result-metadata \
         --sql 'SELECT 1 > 0 AS value' \
         | jq -e '.records[0][0].booleanValue == true'
+
+        aws rds-data execute-statement \
+        --endpoint-url 'http://localhost:8080' \
+        --database 'test' \
+        --resource-arn $RDS_DATA_API_CLIENT_RESOURCE_ARN \
+        --secret-arn $RDS_DATA_API_CLIENT_SECRETARN \
+        --include-result-metadata \
+        --sql $'SELECT \'hello\'::bytea as value' \
+        | jq -e '.records[0][0].blobValue == "aGVsbG8="'
     fi
 
     # select boolean (mysql)
@@ -72,6 +81,15 @@ function test {
         --include-result-metadata \
         --sql 'SELECT 1 > 0 AS value' \
         | jq -e '.records[0][0].longValue == 1'
+
+        aws rds-data execute-statement \
+        --endpoint-url 'http://localhost:8080' \
+        --database 'test' \
+        --resource-arn $RDS_DATA_API_CLIENT_RESOURCE_ARN \
+        --secret-arn $RDS_DATA_API_CLIENT_SECRETARN \
+        --include-result-metadata \
+        --sql $'SELECT BINARY \'hello\' as value' \
+        | jq -e '.records[0][0].blobValue == "aGVsbG8="'
     fi
 
     aws rds-data execute-statement \

